src/umpire/op/HostReallocateOperation.cpp

```diff
@@ -31,20 +33,24 @@ void HostReallocateOperation::transform(
 {
   auto allocator = dst_allocation->m_strategy;
 
-  delete ResourceManager::getInstance().deregisterAllocation(src_ptr);
+  ResourceManager::getInstance().deregisterAllocation(src_ptr);
 
   *dst_ptr = ::realloc(src_ptr, length);
 
+  if (*dst_ptr == src_ptr) {
+    dst_allocation->m_size = length;
```

Comment:
Does this also update the registered allocation of the Inspector with the new length?